lib/Page.js

```diff
@@ -71,6 +80,38 @@ Page.prototype.collectIncludedFiles = function (dependencies) {
   });
 };
 
+/**
+ * Records the front matter data of the current page into this.frontMatter
+ * @param includedPage a page with its dependencies included
+ */
+Page.prototype.collectFrontMatter = function (includedPage) {
+  return new Promise((resolve) => {
+    const $ = cheerio.load(includedPage);
+    const frontMatter = $('frontmatter');
+    const includedFrontMatter = $('frontmatter div');
+    if (frontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
+      let frontMatterData = '';
+      if (includedFrontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
+        frontMatterData = $(includedFrontMatter).get(FIRST_ELEMENT_INDEX).children[FIRST_ELEMENT_INDEX].data;
+      } else {
+        frontMatterData = frontMatter.get(FIRST_ELEMENT_INDEX).children[FIRST_ELEMENT_INDEX].data;
+      }
+      const formattedMatter = `${FRONT_MATTER_FENCE + NEW_LINE}${frontMatterData}${FRONT_MATTER_FENCE}`;
+      // Parse front matter data
+      const parsedData = fm(formattedMatter);
+      parsedData.attributes.src = this.pageSrc;
+      parsedData.attributes.title = this.pageTitlePrefix + parsedData.attributes.title;
+      this.frontMatter = parsedData.attributes;
```

Comment:
Do we need to add `src` and `title` to `parsedData.attributes` here? I think we can use a new object instead.

Reply:
Hm, we could. So a new object that has the fields from `parsedData.attributes` and `title` and `src`? It does seem like more work though, since we're essentially cloning `parsedData.attributes`